colorful/utils.py

```diff
@@ -17,6 +18,17 @@ def hex_to_rgb(value):
     valid RGB channel triplet.
     """
     value = value.lstrip('#')
-
     length = len(value)
-    return tuple(int(value[i:i + int(length / 3)], 16) for i in range(0, length, int(length / 3)))
+
+    check_hex(value, length)
+
+    step = int(length / 3)
+    return tuple(int(value[i:i+step], 16) for i in range(0, length, step))
+
+
+def check_hex(value, length):
+    if re.search(r'[g-z]', value, re.I):
```

Comment:
I'd check if all characters are valid instead of one invalid. 
In this implementation `......` would be treates as valid.